### PR TITLE
Implementation of queue using stack & vice-versa

### PR DESCRIPTION
**Pull request by Himanshu44626748**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Implementation of queue using stack and stack using queue in java.

### Checklist

- [✔ ] I've read the contribution guidelines.
- [✔] I've checked the issue list before deciding what to submit.
- [✔] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes for #412 issue

### CODE REVIEW

**Comment by Himanshu44626748** [comment]
@siddharth25pandey @Kajol-Kumari can you please review my PR

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Himanshu44626748 Please do required modifications
* For more better user friendly interface, take input from the user.
* Display relevant success message after each operation.
*  Make a count function to count total number of elements present at that time.

File: Java/ds/QueueandStack.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+		
+		//Printing dequeue items from dequeue function from Qs class
+		System.out.print(obq.dequeue()+" ");
+		System.out.print(obq.dequeue()+" ");
+		System.out.print(obq.dequeue()+" ");
+		System.out.print(obq.dequeue()+" ");
+		System.out.println(obq.dequeue()+" ");
+		
+		//Calling push function from Sq class to push an item into a stack
+		obs.push(1);
```
Comment (Kajol-Kumari) [inline Java/ds/QueueandStack.java]: Please take the input from user instead of hardcoding it .

File: Java/ds/QueueandStack.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+}
+
+
+
+/*Output:
+
+Implementation of queue using stack
+1 2 3 4 5 
+Implementation of stack using queue
+5 4 3 2 1  */
```
Comment (Kajol-Kumari) [inline Java/ds/QueueandStack.java]: Please add one extra blank line at the end of the file.

File: Java/ds/QueueandStack.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+		return temp;
+	}
+}
+
+//QueueandStack class with main function to operate Qs and Sq class
+public class QueueandStack {
+	public static void main(String args[])
+	{
+		Sq obs = new Sq(5);		//Creating an object of Sq class
+		Qs obq = new Qs(5);		//Creating an object of Qs class
```
Comment (Kajol-Kumari) [inline Java/ds/QueueandStack.java]: Please take the size inputs from user.

File: Java/ds/QueueandStack.java [hunk 4]
```diff
@@ -0,0 +1,219 @@
+//Qs class for implementing queue using two stack with costly dequeue() 
+class Qs {
+	int s1[];			//First stack
+	int s2[];			//Second stack
+	int size;			//size of queue
+	int topS1;			//To maintain top of stack 1
+	int topS2;			//To maintain top of stack 2
+	int count;  		//To count number of items in queue
+	public Qs(int s)
```
Comment (Kajol-Kumari) [inline Java/ds/QueueandStack.java]: Please provide a more meaningful name maybe `QueueUsingStack` ?

File: Java/ds/QueueandStack.java [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+		{
+			temp = popS2();
+			pushS1(temp);
+		}
+		return dq;
+	}
+}
+
+//Sq class to implement a stack using two queue with costly push()
+class Sq {
```
Comment (Kajol-Kumari) [inline Java/ds/QueueandStack.java]: Can you please give the class a more meaningful name like `StackUsingQueue`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @Himanshu44626748 please incorporate the suggested changed and it will be good to go :)

**Comment by Himanshu44626748** [comment]
@siddharth25pandey @Kajol-Kumari I have modified my code as per your instruction. Please review it.

**Review by siddharth25pandey — APPROVED** [review]
@Himanshu44626748 Looks fine to me
Able to merge
Thanks!

**Review by iamrajiv — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
Perfect!
This can be merged now :)

**Comment by iamrajiv** [comment]
@all-contributors please add @Himanshu44626748 for code

**Pull request merged**